### PR TITLE
Support alternative thumb file formats

### PR DESCRIPTION
**Pull request by distantnative**
### Feature
- Support for alternative thumb file formats (e.g. `webp` or `avif`)
- Either set as global option `'thumbs.format' => 'webp'` or pass as option:
```php
$image->thumb(['width' => 200, 'format' => 'webp'])

$image->srcset([
  ['width' => 200, 'format' => 'webp'],
  ['width' => 200, 'format' => 'jpg']
])
```

- GDLib/SimpleImage driver currently only supports `webp`, ImageMagick supports `webp` and `avif`

## Related issues/ideas

- Implements https://kirby.nolt.io/160

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] CI checks pass

## When merging

- [x] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)
- [x] Add changes to release notes draft in Notion

### CODE REVIEW

File: src/Cms/FileModifications.php [hunk 1]
```diff
@@ -191,6 +191,13 @@ public function thumb($options = null)
             return $this;
         }
 
+        // fallback to global config options
```
Comment (distantnative) [inline src/Cms/FileModifications.php:194]: I'm unsure myself if this is at the right location.
First I had it only in the `thumb` core component, but then the `file::version` component creates a wrong filename with wrong extension. So we somehow need to pass that option to `file::version` as well.

**Review by bastianallgeier — COMMENTED** [review]
It looks really good in general. But it also shows how weird this entire setup is. The elephant in the room is probably if we should rip this entire shit out and redo it from scratch, or if we should just get this feature done.

**Comment by distantnative** [comment]
I would redo it eventually, but not sure if 3.6 is feasible for that.

Given how little changes are needed with the current setup, if we can't do the big step in 3.6, I would still do this smaller one given the demand for this support.

**Review by bastianallgeier — DISMISSED** [review]
If we redo it later anyway, I wouldn't bother about the position of the format fallback in your code. I think the rest really looks good to me. Could you take care of fixing the tests?

**Review by bastianallgeier — DISMISSED** [review]

**Comment by distantnative** [comment]
Hmmm... not sure why my one test is actually not asserting anything

**Comment by distantnative** [comment]
@bastianallgeier got it, but need another approval

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**